sdk/python/kfp/components/tasks_group.py

```diff
@@ -143,15 +143,15 @@ class Condition(TasksGroup):
     definition.
 
     Args:
-        condition: The condition expression. Can be constructed using constants or outputs from upstream tasks.
+        condition: A comparative expression that evaluates to True or False. At least one of the operands must be an output from an upstream task.
```

Comment:
> At least one of the operands must be an output from an upstream task

That shouldn't be the case, do you find it otherwise, that would be a bug then.

Reply:
Good catch. I believe this should read "...an output from an upstream task or a pipeline parameter". This seems like desired behavior to me, since the only other option is a static value which is known at compile-time, defeating the purpose of the conditional expression.

Given the following component:

```python
@dsl.component
def my_component() -> int:
    return 1
```

The following two pipelines are successful:

```python
@dsl.pipeline()
def my_pipeline(num: int = 1):
    with dsl.Condition(num == 1):
        my_component()
```

```python
@dsl.pipeline()
def my_pipeline():
    task = my_component()
    with dsl.Condition(task.output == 1):
        my_component()
```

But this pipeline is not:


```python
@dsl.pipeline()
def my_pipeline():
    with dsl.Condition(1 == 1):
        my_component()
```

Reply:
I tend to think it's a bug that `with dsl.Condition(1 == 1):` doesn't work, although it probably doesn't provide any practical value. So I'm okay if we want to update the docstring instead of fixing the bug.

/lgtm

Reply:
Agreed that making `dsl.Condition` robust to this is a better case. I'll merge this PR as it's low-hanging fruit to make the codebase consistent, but will create a bug indicating this.